autora_closed_loop_demo/researcher_hub/autora_workflow.py

```diff
@@ -81,140 +95,395 @@ def theorist_on_state(experiment_data, variables):
 
 
 @on_state()
-def experimentalist_on_state(variables, num_samples):
-    return Delta(conditions=pool(variables, num_samples))
-
-
-# ** Experiment Runner ** #
-# We will run our experiment on firebase and need credentials. You will find them here:
-# (https://console.firebase.google.com/)
-#   -> project -> project settings -> service accounts -> generate new private key
-
-firebase_credentials = {
-}
-
-# simple experiment runner that runs the experiment on firebase
-experiment_runner = firebase_runner(
-    firebase_credentials=firebase_credentials,
-    time_out=100,
-    sleep_time=5)
+def experimentalist_on_state(variables, num_samples, experimentalist=pool):
+    return Delta(conditions=experimentalist(variables, num_samples))
 
 
 # Again, we need to wrap the runner to use it on the state. Here, we send the raw conditions.
 @on_state()
-def runner_on_state(conditions):
+def runner_on_state(conditions, experiment_runner: callable):
     # Here, we convert conditions into sweet bean code to send the complete experiment code
     # directly to the server
 
-    res = []
-
-    for idx, c in conditions.iterrows():
-        i_1 = c['S1']
-        i_2 = c['S2']
-        # get a timeline via sweetPea
-        # can also do different timelines
-        timeline = trial_sequences(i_1, i_2, 10)[0]
-        # get js code via sweetBeaan
-        js_code = stimulus_sequence(timeline, i_1, i_2)
-        res.append(js_code)
-    
+    coherence_ratios_list = list(conditions["coherence_ratio"])
+    motion_directions_list = list(conditions["motion_direction"])
     conditions_to_send = conditions.copy()
-    conditions_to_send['experiment_code'] = res
-    # upload and run the experiment, sending is to firestore database as conditions
-    data_raw = experiment_runner(conditions_to_send) # returns observations for each condition as jsPsych data
+
+    # global training_seq
+    # experiment_timeline = trial_sequences(coherence_ratios_list, motion_directions_list, all_items_in_one_trial=True)[0]
+    # js_code = stimulus_sequence(experiment_timeline, training_timeline=training_seq, to_html=False)
+
+    experiment_seq = trial_sequences(
+        coherence_ratios=[100, 50, 0],
```

Comment:
if we end up adjusting it based on the comment above, then adjust this needs to be adjusted accordingly here too